pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -689,27 +694,88 @@ func CredentialHelpers(sys *types.SystemContext) ([]string, error) {
 	return config.partialV2.CredentialHelpers, nil
 }
 
-// refMatchesPrefix returns true iff ref,
+// refMatchingSubdomainPrefix returns the length of ref without the repo name
+// iff ref, which is a registry, repository namespace, repository or image reference (as formatted by
+// reference.Domain(), reference.Named.Name() or reference.Reference.String()
+// — note that this requires the name to start with an explicit hostname!),
+// matches a Registry.Prefix value containing wildcarded subdomains in the
+// format: *.example.com. Wildcards are only accepted at the beginning, so
+// other formats like example.*.com will not work. Wildcarded prefixes also
+// cannot contain port numbers or namespaces in them.
+func refMatchingSubdomainPrefix(ref, prefix string) int {
+	origRef := ref
+
+	if strings.Contains(prefix[2:], "/") || strings.Contains(prefix[2:], ":") || strings.Contains(prefix[2:], "@") {
+		panic("Invalid wildcarded prefix")
```

Comment:
If we want to validate prefixes, we must do that when _loading_ the config and return an error.  A library should not panic unless there is absolutely no other way to recover or immediate harm to the machine or the user.

Reply:
The error message should include the reason (i.e., what the invalid part is) and a potential resolution (i.e., what a valid format would be).

Reply:
I'll move it to loadConfig, thanks